src/nbl/asset/utils/CIESProfileParser.cpp

```diff
@@ -167,48 +171,37 @@ bool CIESProfileParser::parse(CIESProfile& result)
     float totalEmissionIntegral = 0.0, nonZeroEmissionDomainSize = 0.0;
     constexpr auto FULL_SOLID_ANGLE = 4.0f * core::PI<float>();
 
-    if (result.symmetry == CIESProfile::ISOTROPIC && result.vAngles.size() == 1) // not sure if any IES provides this kind of distribution but it should be possible to do so meaning fixed candela value for each direction
+    for (size_t i = 0; i < result.hAngles.size() - 2; i++)
     {
-        const auto candelaValue = result.getCandelaValue(0, 0);
-        constexpr auto solidAngle = FULL_SOLID_ANGLE; // spans whole sphere area
+        const float dPhiRad = result.symmetry != CIESProfile::ISOTROPIC ? (hAngles[i + 1] - hAngles[i]) : core::PI<float>() * 2.0f;
 
-        totalEmissionIntegral = candelaValue * solidAngle;
-        nonZeroEmissionDomainSize = solidAngle;
-    }
-    else
-        for (size_t i = 0; i < result.hAngles.size(); i++)
-            for (size_t j = 0; j < result.vAngles.size(); j++)
+        for (size_t j = 0; j < result.vAngles.size() - 2; j++)
+        {
+            const auto candelaValue = result.getCandelaValue(i, j);
+
+            if (result.maxCandelaValue < candelaValue)
```

Comment:
where do you initialize to 0 , if you don't initialize to 0, its not given that `<` will pass on first try

Reply:
`IES_STORAGE_FORMAT maxCandelaValue = {}; in header of the parser